src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs

```
@@ -216,7 +215,7 @@ public virtual SelectionRules SelectionRules
 
             if ((prop = props["Size"]) is not null && !prop.IsReadOnly)
             {
-                if (AutoResizeHandles && Component != _host.RootComponent)
+                if (AutoResizeHandles && Component != _host?.RootComponent)
```

It looks like this is where our tests were originally expecting NRE. Should this be changed to throw `InvalidOperationException` instead of continuing on when `_host = null`? Thoughts @Tanya-Solyanik , @JeremyKuhne ?

---

In this case I'd add a `Debug.Assert` and keep the `?.`. Presumption is that this probably should not happen, but there is no real bad in returning rules in this case. You'll need to add a no assert context to the test.